src/features/distribute/components/ConfirmDistributionDialog.tsx

```diff
@@ -36,35 +42,71 @@ export function ConfirmDistributionDialog({
     );
   }, [distribution]);
 
-  const amountDiff = (balance ?? 0n) - amounts.reduce((sum, x) => sum + x, 0n);
+  const balanceBigInt = BigInt(balance ?? 0);
+  const totalAmounts = amounts.reduce((sum, x) => sum + x, BigInt(0));
+  const isPoolUnderfunded = totalAmounts > balanceBigInt;
+  const amountDiff = isPoolUnderfunded
+    ? totalAmounts - balanceBigInt
+    : balanceBigInt - totalAmounts;
 
-  return (
-    <Dialog
-      isOpen={distribution.length > 0}
-      size="sm"
-      title="Confirm distribution"
-      onOpenChange={onOpenChange}
-    >
+  const renderFundingMessage = () => (
+    <div>
       <div className="mb-4">
-        This will distribute the pools funds to the payout addresses according
-        to the table.
+        Not enough funds in the pool to distribute. Please fund the pool first.
+      </div>
+      <div className="mb-4 flex flex-col items-center">
+        <h3 className="mb-2 text-sm font-semibold uppercase tracking-widest text-gray-500">
+          <div>Amount Required</div>
+        </h3>
+        <div
+          className={cn("text-2xl", { ["text-red-600"]: isPoolUnderfunded })}
+        >
+          {formatUnits(amountDiff, token.decimals)}
+        </div>
       </div>
+      <div className="space-y-1">
+        <IconButton
+          disabled={isFunding || !poolId}
+          icon={isFunding ? Spinner : null}
+          className={"w-full"}
+          variant="primary"
+          onClick={() => {
+            if (poolId) {
+              fundPool?.(
+                { poolId, amount: amountDiff },
+                { onSuccess: () => setPoolFunded(true) },
+              );
+            }
+          }}
+        >
+          {isPending ? "Funding in progress..." : "Fund Pool"}
+        </IconButton>
+        <Button className={"w-full"} onClick={onOpenChange}>
+          Cancel
+        </Button>
+      </div>
+    </div>
+  );
 
+  const renderDistributionMessage = () => (
```

Comment:
can you create a component for this? better than a render function, unless is a temporary thing for testing